src/webgpu/shader/execution/expression/expression.ts

```diff
@@ -418,6 +422,27 @@ struct Output {
 @group(0) @binding(0) var<storage, read_write> outputs : array<Output, ${count}>;`;
 }
 
+/**
+ * Helper that returns the WGSL to declare the values array for a shader
+ */
+function wgslValuesArray(
+  parameterTypes: Array<Type>,
+  resultType: Type,
+  cases: CaseList,
+  expressionBuilder: ExpressionBuilder
+): string {
+  // f64/AbstractFloat values cannot be stored in an array
+  if (parameterTypes.some(ty => scalarTypeOf(ty).kind === 'f64')) {
+    return '';
+  }
+  return `
+const values = array(
+  ${cases
+    .map(c => toStorage(resultType, expressionBuilder(map(c.input, v => v.wgsl()))))
```

Comment:
I can't recall why we go `toStorage` for this `const` array. It seems like it would be generally simpler to fill this with the non-storage values, which supports both `bool`s and abstract-numerics.